server/entity/splash_potion.go

```diff
@@ -137,24 +148,24 @@ func (s *SplashPotion) Tick(w *world.World, current int64) {
 					splashed.AddEffect(effect.New(eff.Type().(effect.LastingType), eff.Level(), dur))
 				}
 			}
-		} else if s.t == potion.Water() {
-			switch result := result.(type) {
-			case trace.BlockResult:
-				pos := result.BlockPosition().Side(result.Face())
-				if w.Block(pos) == fire() {
-					w.SetBlock(pos, air(), nil)
-				}
-
-				for _, f := range cube.HorizontalFaces() {
-					if h := pos.Side(f); w.Block(h) == fire() {
-						w.SetBlock(h, air(), nil)
-					}
-				}
-			case trace.EntityResult:
-				// TODO: Damage endermen, blazes, striders and snow golems when implemented and rehydrate axolotls.
+		}
+		// Splashing Entities and Blocks
+		switch result := result.(type) {
+		case trace.BlockResult:
+			// we first check to see if it splashed an empty block that's splashable
+			pos := result.BlockPosition().Side(result.Face())
+			block := w.Block(pos)
+			if splashable, ok := block.(SplashableBlock); ok && block.Model() == emptyModel() {
```

Comment:
Would use a type assertion for the block model here, you can get rid of the `emptyModel` function

Reply:
done :smile: